components/camel-wasm/src/test/java/org/apache/camel/language/wasm/WasmLanguageTest.java

```diff
@@ -0,0 +1,98 @@
+ * limitations under the License.
+ */
+package org.apache.camel.language.wasm;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.FluentProducerTemplate;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.apache.camel.util.StringHelper;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.as;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class WasmLanguageTest {
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "transform@functions.wasm",
+            "transform@file:{{wasm.resources.path}}/functions.wasm"
+    })
+    public void testLanguage(String expression) throws Exception {
+        try (CamelContext cc = new DefaultCamelContext()) {
+
+            FluentProducerTemplate pt = cc.createFluentProducerTemplate();
+
+            cc.addRoutes(new RouteBuilder() {
+                @Override
+                public void configure() throws Exception {
+                    from("direct:in")
+                            .transform()
+                            .wasm(
+                                    StringHelper.before(expression, "@"),
+                                    StringHelper.after(expression, "@"));
+                }
+            });
+            cc.start();
+
+            Exchange out = pt.to("direct:in")
+                    .withHeader("foo", "bar")
+                    .withBody("hello")
+                    .request(Exchange.class);
+
+            assertThat(out.getMessage().getHeaders())
+                    .containsEntry("foo", "bar");
+            assertThat(out.getMessage().getBody(String.class))
+                    .isEqualTo("HELLO");
+
+        }
+    }
+
+    @Test
+    public void testLanguageFailure() throws Exception {
+        try (CamelContext cc = new DefaultCamelContext()) {
+
+            FluentProducerTemplate pt = cc.createFluentProducerTemplate();
+
+            cc.addRoutes(new RouteBuilder() {
+                @Override
+                public void configure() throws Exception {
+                    from("direct:in")
+                            .transform()
+                            .wasm("transform_err", "functions.wasm");
+                }
+            });
+            cc.start();
+
+            Exchange out = pt.to("direct:in")
+                    .withHeader("foo", "bar")
+                    .withBody("hello")
+                    .request(Exchange.class);
+
+            assertThat(out.getException())
+                    .isNotNull()
+                    .hasCauseInstanceOf(RuntimeException.class)
+                    .extracting(Throwable::getCause, as(InstanceOfAssertFactories.THROWABLE))
+                    .hasMessage("this is an error");
```

Comment:
One thing I noticed: we need to add a `log4j2.properties` file to the `test/resources` directory. This is causing the test to leak to stdout. 

Something like this should do the trick:

```
appender.out.type = File
appender.out.name = out
appender.out.fileName = target/camel-wasm-test.log
appender.out.layout.type = PatternLayout
appender.out.layout.pattern = %d [%-15.15t] %-5p %-30.30c{1} - %m%n
appender.stdout.type = Console
appender.stdout.name = stdout
appender.stdout.layout.type = PatternLayout
appender.stdout.layout.pattern = %d [%-15.15t] %-5p %-30.30c{1} - %m%n

rootLogger.level = WARN
rootLogger.appenderRef.out.ref = out

logger.camel.name=org.apache.camel
logger.camel.level=INFO

logger.camelWasm.name=org.apache.camel.component.wasm
logger.camelWasm.level=DEBUG
```

Reply:
done